src/compiler/checker.ts

```diff
@@ -2766,7 +2766,9 @@ namespace ts {
         }
 
         function getBaseTypeNodeOfClass(type: InterfaceType): ExpressionWithTypeArguments {
-            return getClassExtendsHeritageClauseElement(<ClassLikeDeclaration>type.symbol.valueDeclaration);
+            let classDeclaration =
+                getDeclarationOfKind(type.symbol, SyntaxKind.ClassDeclaration) || type.symbol.valueDeclaration;

```

Comment:
But in the normal case (not the ambient declaration case), classes are required to be before namespaces already. Changing the binder/checker-merge to merge `valueDeclaration` from non-namespaces just makes ambient declarations behave the same way as the normal case, even if the namespace comes first. 

Besides, other uses of `valueDeclaration` very likely also assume that it is not a namespace since namespaces normally come last.